lib/protobuf/generators/file_generator.rb

```diff
@@ -49,8 +50,14 @@ def compile
       end
 
       def unknown_extensions
-        @unknown_extensions ||= @extension_fields.reject do |k, _|
-          @known_messages.include?(k)
+        @unknown_extensions ||= @extension_fields.map do |message_name, fields|
+          message_klass = modulize(message_name).safe_constantize
+          next [message_name, fields] unless message_klass
```

Comment:
Would make this an if statement. What you have right now is a little confusing since using `next` in that fashion isn't that common.